python_src/src/lamp_py/migrations/versions/performance_manager_prod/001_5d9a7ee21ae5_initial_prod_schema.py

```diff
@@ -0,0 +1,966 @@
+        CREATE TRIGGER insert_into_feed_info BEFORE INSERT ON static_feed_info 
+        FOR EACH ROW EXECUTE PROCEDURE insert_feed_info();
+    """
+    op.execute(create_insert_into_feed_info_trigger)
+
+    create_red_is_ashmont_branch_func = """
+        CREATE OR REPLACE FUNCTION red_is_ashmont_branch(p_trip_id varchar, p_fk_ts int) RETURNS boolean AS $$ 
+        DECLARE
+            is_ashmont boolean;
+        BEGIN 
+            SELECT
+                true
+            INTO
+                is_ashmont
+            FROM 
+                static_stop_times sst
+            WHERE 
+                sst.trip_id = p_trip_id
+                AND sst.static_version_key = p_fk_ts
+                AND sst.stop_id IN ('70087', '70088', '70089', '70090', '70091', '70092', '70093', '70094', '70085', '70086')
+            LIMIT 1
+            ;
+
+            IF FOUND THEN
+                RETURN true;
+            ELSE
+                RETURN false;
+            END IF;
+        END;
+        $$ LANGUAGE plpgsql;
+    """
+    op.execute(create_red_is_ashmont_branch_func)
+
+    create_red_is_braintree_branch_func = """
+        CREATE OR REPLACE FUNCTION red_is_braintree_branch(p_trip_id varchar, p_fk_ts int) RETURNS boolean AS $$ 
+        DECLARE	
+            is_braintree boolean;
+        BEGIN 
+            SELECT
+                true
+            INTO
+                is_braintree
+            FROM 
+                static_stop_times sst
+            WHERE 
+                sst.trip_id = p_trip_id
+                AND sst.static_version_key = p_fk_ts
+                AND sst.stop_id IN ('70097', '70098', '70099', '70100', '70101', '70102', '70103', '70104', '70105', '70095', '70096')
+            LIMIT 1
+            ;
+
+            IF FOUND THEN
+                RETURN true;
+            ELSE
+                RETURN false;
+            END IF;
+        END;
+        $$ LANGUAGE plpgsql;
+    """
+    op.execute(create_red_is_braintree_branch_func)
+
+    create_insert_static_trips_branch_trunk = """
+        CREATE OR REPLACE FUNCTION insert_static_trips_branch_trunk() RETURNS TRIGGER AS $$ 
+        BEGIN 
+            IF NEW.route_id ~ 'Green-' THEN
+                NEW.branch_route_id := NEW.route_id;
+                NEW.trunk_route_id := 'Green';
+            ELSEIF NEW.route_id = 'Red' THEN
+                IF red_is_braintree_branch(NEW.trip_id, NEW.static_version_key) THEN
+                    NEW.branch_route_id := 'Red-B';
+                ELSEIF red_is_ashmont_branch(NEW.trip_id, NEW.static_version_key) THEN
+                    NEW.branch_route_id := 'Red-A';
+                ELSE
+                    NEW.branch_route_id := null;
+                END IF;
+                NEW.trunk_route_id := NEW.route_id;
+            ELSE
+                NEW.branch_route_id := null;
+                NEW.trunk_route_id := NEW.route_id;
+            END IF;
+            RETURN NEW;
+        END;
+        $$ LANGUAGE plpgsql;
+    """
```

Comment:
does it make sense to have this command as a stored as a string and insert it into these commands since its largely the same between the static and rt versions of this?

might not be worth it, but its always a code smell to see duplication.

Reply:
The rt vs static functions are slightly different enough that they couldn't be stored as the same string. The whole inserting giant strings with alembic into the RDS in general feels pretty messy, but I'm not sure if a better solution exists. 

It could be worth extracting sections of the command, like the list of stop_ids and storing those as a string, to be inserted. 

Might be worth having a quick call to discuss options.